### PR TITLE
dpdk: add warning in configuration phase and minor DPDK refactoring v1

### PR DESCRIPTION
**Pull request by lukashino**
Redmine ticket: 
https://redmine.openinfosecfoundation.org/issues/6739

Describe changes:
- SCEnters added
- refactor code to smaller functions
- rewrite log message to be more consistent
- add a user warning n the change of RX/TX descriptors - on changes caused by the NIC adjustment

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 101 | 111 | 109.9% | 

Pipeline 18389

File: src/runmode-dpdk.c [hunk 1]
```diff
@@ -1550,7 +1578,7 @@ static void *ParseDpdkConfigAndConfigureDevice(const char *iface)
     if (retval < 0) { // handles both configure attempts
         iconf->DerefFunc(iconf);
         if (rte_eal_cleanup() != 0)
-            FatalError("EAL cleanup failed: %s", strerror(-retval));
+            FatalError("EAL cleanup failed (err: %s)", strerror(-retval));
```
Comment (victorjulien) [inline src/runmode-dpdk.c:1581]: I don't see this as an improvement. The old scheme is as clear, while being less verbose. Would rather unify on that. This also matches use of `strerror(errno)` elsewhere in the code

Comment (victorjulien) [reply src/runmode-dpdk.c:1581]: same for the rest of the PR

**Review by victorjulien — CHANGES_REQUESTED** [review]
not agreeing with the output changes

**Comment by lukashino** [comment]
continues in #10449

**Pull request closed without merging**